### ISSUE TITLE
Mac osx 64 NWJS game unexpectedly quitting

### ISSUE THREAD

**Issue by cloudnovel** [nwjs/nw.js#6606]
======== TEMPLATE BUG FORM ========

NWJS Version : 66.0.3359.139
Operating System : Mac

### Expected behavior

<!- What do you think should happen? ->
For some reason the Mac visual novel game i build using NWJS Builder to generate it through npm only works on my Mac, but when I upload it to google drive and let friends download the NWJS game it unexpectedly quits. At first it was just saying that the it was an unrecognized developer, but my friends went into the security settings and disabled that and it then unexpectedly quits and crashes.
### Actual behavior

<!- What actually happens? ->
On the Mac that I build the NWJS game using my Mac, it plays and opens just fine. However, when uploaded to google drive and my friends try to open it on their macbooks it unexpectedly crashes. 

### How to reproduce

Try these games here i uploaded to google drive, they ALL crash:
https://drive.google.com/open?id=1CSSYsNb7V6T0h_Hp8QyJ0lVyTg74Uo6S
https://drive.google.com/open?id=1HVJXz7-zIBkcR1h3hhtV6M4Mhg5jOmsr
https://drive.google.com/open?id=18To1wx-iHCPGACWUB_4N13YyDdjdGu5X
https://drive.google.com/open?id=15r8lVzCEu4Iv-oA2HMA5HNqWJj-aRDQS

All of them unexpectedly quit. I don't know what's wrong, all the windows NWJS games work just fine, just the Mac ones don't work.

I also went to the apple store to test it out on fresh mac computers and all of the games unexpectedly crash. Yes I have already tested them in new environments. 

Please help!!
<!- 

- Detail a step by step guide to reproduce the issue
- A copy of your code would help. Include: JS, HTML. 
- If you have a crash report, copy it in. 
- Test your steps on a fresh environment to see if you still have an issue 
- Post!

->

**Comment by Christywl** [comment]
I download and try it on Mac:
```
 $ ./VNosx-x64/VisualNovel.app/Contents/MacOS/nwjs 
dlopen /Users/test/nwjs/bugs/6606/./VNosx-x64/VisualNovel.app/Contents/MacOS/../Versions/66.0.3359.139/nwjs Framework.framework/nwjs Framework: dlopen(/Users/test/nwjs/bugs/6606/./VNosx-x64/VisualNovel.app/Contents/MacOS/../Versions/66.0.3359.139/nwjs Framework.framework/nwjs Framework, 257): image not found
Abort trap: 6
```

**Comment by Blatman** [comment]
FWIW I tried the first link (relentless) with 0.30.3-sdk and it failed - I noticed that the symlinks in the nwjs Framework.framework package are all unix executables instead. I tried the app.nw from the faulty package in a good build package and it works fine and if I use the nwjs Framework.framework package from a standard good build in the faulty package it also works fine. The faulty package appears to have been built with nwjs-builder.
Hope this helps.

**Comment by cloudnovel** [comment]
@Blatman Which one is the faulty package, sorry I am new to this. Specifically what do I replace, do I replace the framework.frame file with another file? Where can I get this other file to replace it with?

**Comment by cloudnovel** [comment]
I just tried this: https://drive.google.com/file/d/1Zg9Bv6W8cLEUpDqrO4bQ3I0G4CY7JkJd/view?usp=sharing

Does this work for you? I'm not on my home computer at the moment so I can't test it. Did I do it right?

**Comment by cloudnovel** [comment]
@Blatman Can you give me a working example of what you said you did, the one you said you replaced the framework.framework? I just need a working example.

**Comment by Blatman** [comment]
@cloudnovel 
Take for example 1st link which is relentless.zip which contains relentless.app
On Mac do right-click - Show package Contents and drill down to nwjs Framework.framework bundle.
(relentless.app>Contents>Versions>66.0.3359.139>nwjs Framework.framework)
Replace nwjs Framework.package in relentless.zip bundle with one from same version of nwjs package that your app uses - downloads are https://dl.nwjs.io/v0.30.3/
Your app specific files are located in - relentless.app>Contents>Resources>app.nw

I suspect that the build process is somehow messing with the framework package - usually these are just copied as is by the build process.

**Comment by cloudnovel** [comment]
@Blatman Okay I did exactly what you said and uploaded this to google drive: https://drive.google.com/file/d/1-I7Q_Wl9rQ8lQ8p5J6BGf4gAtaDyXoY_/view?usp=sharing

Does it work for you??

**Comment by Blatman** [comment]
@cloudnovel 
Ah no - still crashes and if I look at the framework package the symlinks are still unix executables.
Again if I use a good nwjs Framework.framework package it is fine.

Plan B would be to take your app.nw folder out of the relentless.app package and insert into a good downloaded nwjs package - it goes in the Resources folder and can be done with cut/paste - don't use any builders for this test.
BTW - if the relentless.app is being created on a Windows platform rather than OSX then be wary - I am making a builder and have found that building OSX on Windows(10) has all sorts of issues with permissions and symlinks.

Cheers.

**Comment by cloudnovel** [comment]
@Blatman Okay I'll go home later on this after noon and try assembling it on my mac. Is there a chance uploading it to google drive messes with the framework?
What I did yesterday was generate it using nw-builder on my mac, but then I went to the mac store and try and open in a new mac environment it crashes. I used nw-builder on Mac to build it, but when I tried to upload it to google drive to let my friends download the game, it crashes/quits. 

Could uploading to google drive also be a possible problem?
